app/models/miq_expression/field.rb

```diff
@@ -29,11 +29,18 @@ def datetime?
     column_type == :datetime
   end
 
+  def reflections
+    klass = model
+    associations.collect do |association|
+      klass.reflection_with_virtual(association).tap { |reflection| klass = reflection.klass }
```

Comment:
I've been batting around this method as well. Not sure the best way to do this.

a. could you move this over to `ar_virtual.rb`?
b. I think this needs to handle invalid associations:

```ruby
klass.try!(:reflection_with_virtual, association).tap { |reflection| klass = reflection.try!(:klass) }
```

I hope you'll come up with a way to not rely so heavily on `try!`